extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -554,23 +554,31 @@ export class AmpAnalytics extends AMP.BaseElement {
       }
     }
 
-    return Promise.all(requestPromises)
-      .then(() => {
-        request = this.addParamsToUrl_(request, params);
-        this.config_['vars']['requestCount']++;
-        const expansionOptions = this.expansionOptions_(event, trigger);
-        return this.variableService_.expandTemplate(request, expansionOptions);
-      })
-      .then(request => {
-        const whiteList = this.isSandbox_ ? SANDBOX_AVAILABLE_VARS : undefined;
-        // For consistency with amp-pixel we also expand any url replacements.
-        return urlReplacementsForDoc(this.element).expandAsync(
-            request, undefined, whiteList);
-      })
-      .then(request => {
-        this.sendRequest_(request, trigger);
-        return request;
-      });
+    return this.checkTriggerEnabled_(trigger, event)
```

Comment:
would it make more sense to check for `enabled` before we do the url expansion. That way we prevent some work that might not be used.